frontend/src/app/components/wp-table/timeline/container/wp-timeline-container.directive.ts

```diff
@@ -416,4 +471,77 @@ export class WorkPackageTimelineTableController extends UntilDestroyedMixin impl
       }
     }
   }
+
+  manageCollapsedGroupHeaderCells(allGroups:GroupObject[],
+                                  collapsedGroupsChange:{[key:string]:boolean}|undefined,
+                                  tableWorkPackages:WorkPackageResource[],
+                                  collapsedGroupsCellsMap:IGroupCellsMap,
+                                  updateAllHeaderCells?:boolean) {
+    if (!collapsedGroupsChange) {
+      return;
+    }
+
+    const collapsedGroupChangesToManage = Object.keys(collapsedGroupsChange).filter(groupKey => {
+      const keyGroupType = groupKey.split('-')[0];
+
+      return this.groupTypesWithHeaderCellsWhenCollapsed.includes(keyGroupType);
+    });
+    let groupsToUpdate:string[];
+
+    if (updateAllHeaderCells) {
+      groupsToUpdate = collapsedGroupChangesToManage;
+    } else {
+      groupsToUpdate = collapsedGroupChangesToManage.filter(groupKey => {
+        const currentGroupCollapsedValue = collapsedGroupsChange[groupKey];
+        const storedGroup = allGroups.find(group => group.identifier === groupKey);
+
+        return storedGroup && storedGroup.collapsed !== currentGroupCollapsedValue;
+      });
+    }
+
+    groupsToUpdate.forEach(groupIdentifier => {
+      const groupIsCollapsed = collapsedGroupsChange[groupIdentifier];
+
+      if (groupIsCollapsed) {
+        this.createCollapsedGroupHeaderCells(groupIdentifier, tableWorkPackages, collapsedGroupsCellsMap);
+      } else {
+        this.removeCollapsedGroupHeaderCells(groupIdentifier, collapsedGroupsCellsMap);
+      }
+    });
+  }
+
+  createCollapsedGroupHeaderCells(groupIdentifier:string, tableWorkPackages:WorkPackageResource[], collapsedGroupsCellsMap:IGroupCellsMap) {
+    this.removeCollapsedGroupHeaderCells(groupIdentifier, collapsedGroupsCellsMap);
+
+    const changedGroupId = groupIdentifier!.split('-').pop();
+    const changedGroupTableWorkPackages = tableWorkPackages.filter(tableWorkPackage => tableWorkPackage.project.id === changedGroupId);
+    const changedGroupWpsWithHeaderCells = changedGroupTableWorkPackages.filter(tableWorkPackage => this.shouldBeShownInCollapsedGroupHeaders(tableWorkPackage) &&
+                                                                                                    (tableWorkPackage.date || tableWorkPackage.startDate));
```

Comment:
As long as we are only displaying milestones which only sport the `date` property, we should not take any work package that has a `startDate` value